docs/cudf/source/developer_guide/library_design.md

```diff
@@ -221,6 +212,7 @@ Conversely, when constructed from a host object,
 The data is then copied from the host object into the newly allocated device memory.
 You can read more about device memory allocation with RMM [here](https://github.com/rapidsai/rmm).
 
+(cythonlayer)=
```

Comment:
So this specific format isn't even just rst, it's [myst](https://myst-parser.readthedocs.io/en/latest/syntax/syntax.html#targets-and-cross-referencing). I'm not sure I follow what you're suggesting, though?

Reply:
This is a very minor thing, in the grand scheme of things, so I don't think it should hold this PR up, but let me try and explain myself.

> So this specific format isn't even just rst, it's [myst](https://myst-parser.readthedocs.io/en/latest/syntax/syntax.html#targets-and-cross-referencing). 

Myst is just markdown-esque syntax for rst (without the nested parsing issues that docutils have), but this expands into the equivalent of `.. _cythonlayer:` I think.

Here you want to be able to link to this section, but with custom anchors there's no guarantee that someone else won't also pick the same anchor name later.

In this particular case, I think we could turn on [auto-generated anchors](https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#auto-generated-header-anchors) and then the link would be `[](developer_guide/library_design.md#the-cython-layer)`

> I'm not sure I follow what you're suggesting, though? Are you saying that we should have a docs page describing (some parts of) the tool that we use to write the docs?

To me, good online documentation takes advantage of online formats and is link-heavy. If I were writing this in latex, there are good editors that can present overview tables of contents and lists of all section, equation, ... links (with searching). So if I need to reference something, I don't have to remember the name of every section label.

As far as I know, the facilities presented by editors for this in any of the many mark-up languages (including myst-markdown) are quite limited. Consequently, finding links to different sections of the docs requires having them all in my brain (and linking occurs less). [With enough roundtoits I might actually write something for emacs to do this]

The autogenerated header names  have some hierarchy in that they are per-file and the files have a structure (which makes finding things easier), but hand-crafted labels are un-namespaced and so if someone else wants to link they can't guess what to look for is all.

Reply:
> Myst is just markdown-esque syntax for rst (without the nested parsing issues that docutils have), but this expands into the equivalent of .. _cythonlayer: I think.

That's a reasonable description of Myst, and you're right, that's how a link like this expands. 

Regarding the rest, I 100% agree that both MD and reST documents tend to link far less than they should because of the difficulty of placing, remembering, and using anchors (especially given the variable syntaxes). I would also love better editors. 

> In this particular case, I think we could turn on [auto-generated anchors](https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#auto-generated-header-anchors) and then the link would be [](developer_guide/library_design.md#the-cython-layer)

I hadn't seen this feature. I think that's an awesome idea.

Reply:
Done.

Reply:
> I would also love better editors.

Might be worth following progress in https://github.com/swyddfa/esbonio